src/PreferencesDialog.blp

```diff
@@ -0,0 +1,94 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.PreferencesDialog preferences_window {
+  title: _("Preferences");
+
+  Adw.PreferencesPage {
+    title: _("General");
+    icon-name: "preferences-system-symbolic";
+
+    Adw.PreferencesGroup {
+      title: _("URL Matchers");
+      description: _("Configure automatic application selection based on URL patterns");
+
+      Adw.ActionRow {
+        title: _("Add URL Matcher");
+        subtitle: _("Create a new rule to automatically open matching URLs");
+        
+        [suffix]
+        Button add_button {
+          icon-name: "list-add-symbolic";
+          tooltip-text: _("Add new URL matcher");
+          valign: center;
+          styles ["flat"]
+        }
+      }
```

Comment:
This shouldn't be a row

Instead implement this a pill button above the ListBox https://developer.gnome.org/hig/patterns/containers/boxed-lists.html#adding-buttons